### ISSUE TITLE
Angular version with Inject true -> wrong dot version

### ISSUE THREAD

**Issue by meriturva** [rfennell/AzurePipelines#1180]
### Azure DevOps Extensions

Other

### Platform

Azure DevOps Services

### Extension Version

2.31.5

### Describe the bug

Just trying to patch environment.prod.ts file for angular project with yaml:

```
- task: VersionAngularFile@2
  displayName: 'Patch Angular versions'
  inputs: 
     Path: '$(Build.SourcesDirectory)/frontend'
     VersionNumber: '$(version)'
     Injectversion: true
     FilenamePattern: 'environment.prod.ts'
```

Note that injectversions is true so:` [Advanced] Inject Version: If true then the build number will be used without regex processing`

but we have a dot at the string end. See logs:
```
Starting: Patch Angular versions
==============================================================================
Task         : Version Angular Environment.TS File
Description  : Applies a version to a named field in a Angular Environment.TS
Version      : 2.31.5
Author       : Black Marble
Help         : Version: 2.31.5. [More Information](https://github.com/rfennell/vNextBuild/wiki/Version-Assemblies-and-Packages-Tasks/)
==============================================================================
Source Directory:  /home/vsts/work/1/s/frontend
Filename Pattern: environment.prod.ts
Version Number/Build Number: 2.0.577-dev
Version Filter to extract build number: \d+\.\d+\.\d+\.\d+
Version Format for JSON File: {1}.{2}.{3}.{4}
Field to update (all if empty): version
Inject Version: true
Output: Version Number Parameter Name: OutputedVersion
Using provided version number directly
Extracted Build Version: 2.0.577-dev
Angular Version Name will be: 2.0.577-dev.
Added file environment.prod.ts via .endswith
Will apply 2.0.577-dev. to 1 files.
Updating the field 'version' version
Existing Tag: version: '1.0.0'
Replacement Tag: version: '2.0.577-dev.'
/home/vsts/work/1/s/frontend/src/environments/environment.prod.ts - version applied
Set the output variable 'OutputedVersion' with the value 2.0.577-dev.
(node:1832) Warning: Use Cipheriv for counter mode of aes-256-ctr
(node:1832) Warning: Use Cipheriv for counter mode of aes-256-ctr
(node:1832) Warning: Use Cipheriv for counter mode of aes-256-ctr
(node:1832) Warning: Use Cipheriv for counter mode of aes-256-ctr
(node:1832) Warning: Use Cipheriv for counter mode of aes-256-ctr
(node:1832) Warning: Use Cipheriv for counter mode of aes-256-ctr
(node:1832) Warning: Use Cipheriv for counter mode of aes-256-ctr
(node:1832) Warning: Use Cipheriv for counter mode of aes-256-ctr
(node:1832) Warning: Use Cipheriv for counter mode of aes-256-ctr
(node:1832) Warning: Use Cipheriv for counter mode of aes-256-ctr
(node:1832) Warning: Use Cipheriv for counter mode of aes-256-ctr
(node:1832) Warning: Use Cipheriv for counter mode of aes-256-ctr
Finishing: Patch Angular versions

```

### Repo Steps

...

### Expected Behavior

Just script has to patch `environment.prod.ts` with version '2.0.577-dev' and not '2.0.577-dev.' (see ending dot).

**Comment by rfennell** [comment]
I will have a look into this

**Comment by rfennell** [comment]
The tasks in this extension have been generated off each other, so the version angular is a specialisation of the version json file. So your comments are correct.

What I have provided is workaround that you can do now, that matches the unit tests I run. But I will revise the `injectversion` logic so this does not occur in the future

**Comment by meriturva** [comment]
Perfect and thanks for your help! I will try that workaround as soon as possible and comment here!

**Comment by rfennell** [comment]
Will be released as 2.32.x

**Comment by meriturva** [comment]
Works like a charm with version: 2.33.16

```
==============================================================================
Task         : Version Angular Environment.TS File
Description  : Applies a version to a named field in a Angular Environment.TS
Version      : 2.33.16
Author       : Black Marble
Help         : Version: 2.33.16. [More Information](https://github.com/rfennell/vNextBuild/wiki/Version-Assemblies-and-Packages-Tasks/)

```

So thanks!
